### PR TITLE
Exchange handle margin mode

### PR DESCRIPTION
**Pull request by samgermain**
This method is in all my recent PRs, and will be in all my soon PRs, so I need it merged in

### CODE REVIEW

**Review by carlosmiei — APPROVED** [review]
Looks good to me

**Comment by kroitor** [comment]
@carlosmiei @samgermain what if we add the omit statement and return the margin mode and the modified params as an array like we do in other similar methods? 

@carlosmiei related PR: https://github.com/ccxt/ccxt/pull/13328

**Comment by carlosmiei** [comment]
@kroitor Yeah I agree, actually I've suggested that as well in a previous PR
image

File: js/base/Exchange.js [hunk 1]
```diff
@@ -2654,4 +2654,19 @@ module.exports = class Exchange {
...
+        /**
+         * @ignore
+         * @method
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[string|undefined, object]} the marginMode in lowercase as specified by params["marginMode"], this.options["marginMode"] or this.options["defaultMarginMode"]
+         */
+        const defaultMarginMode = this.safeString2 (this.options, 'marginMode', 'defaultMarginMode');
+        const methodOptions = this.safeValue (this.options, methodName, {});
+        const methodMarginMode = (typeof methodOptions === 'string') ? methodOptions : this.safeString2 (methodOptions, 'marginMode', 'defaultMarginMode', defaultMarginMode);
+        const marginMode = this.safeStringLower (params, 'marginMode', methodMarginMode);
```
Comment (ttodua) [inline js/base/Exchange.js:2668]: 2667th line is incorrect imho. we should no longer support `string` moreover for newly implemented properties. the `string` is only supported for backward-compatibility and it only held value of  **market-type**  when method had a string value in options. so, if a method has string value, we should ignore that totally. So, code would be like this:
```
        const methodOptions = this.safeValue (this.options, methodName, {});
        const methodMarginMode = (typeof methodOptions === 'string') ? defaultMarginMode : this.safeString2 (methodOptions, 'marginMode', 'defaultMarginMode', defaultMarginMode);
        const marginMode = this.safeStringLower (params, 'marginMode', methodMarginMode);
```

File: js/base/Exchange.js [hunk 2]
```diff
@@ -2654,4 +2654,19 @@ module.exports = class Exchange {
...
+        params = this.omit (params, 'marginMode');
```
Comment (ttodua) [inline js/base/Exchange.js:2669]: maybe we should check `defaultMarginMode`  in params too, like https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L2816

File: js/base/Exchange.js (hunk 1 above)
Comment (ttodua) [inline js/base/Exchange.js:2668]: maybe we should check `defaultMarginMode`  in params too, like https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L2816

File: js/base/Exchange.js [hunk 3]
```diff
@@ -2654,4 +2654,21 @@ module.exports = class Exchange {
...
+        /**
+         * @ignore
+         * @method
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[string|undefined, object]} the marginMode in lowercase as specified by params["marginMode"], this.options["marginMode"] or this.options["defaultMarginMode"]
+         */
+        const defaultMarginMode = this.safeString2 (this.options, 'marginMode', 'defaultMarginMode');
+        const methodOptions = this.safeValue (this.options, methodName, {});
+        const methodMarginMode = (typeof methodOptions === 'string') ? methodOptions : this.safeString2 (methodOptions, 'marginMode', 'defaultMarginMode', defaultMarginMode);
+        const marginMode = this.safeStringLower (params, 'marginMode', methodMarginMode);
```
Comment (ttodua) [inline js/base/Exchange.js:2668]: 2667th line is incorrect imho. we should no longer support `string` moreover for newly implemented properties. the `string` is only supported for backward-compatibility and it only held value of  **market-type**  when method had a string value in options. so, if a method has string value, we should ignore that totally. So, code would be like this:

```suggestion
        const methodOptions = this.safeValue (this.options, methodName, {});
        const methodMarginMode = (typeof methodOptions === 'string') ? methodOptions : this.safeString2 (methodOptions, 'marginMode', 'defaultMarginMode', defaultMarginMode);
        const marginMode = this.safeStringLower (params, 'marginMode', methodMarginMode);
```

File: js/base/Exchange.js (hunk 1 above)
Comment (samgermain) [reply js/base/Exchange.js:2668]: we haven't historically, but we could. @kroitor please give your input

**Pull request merged**